woocommerce/woocommerce-gateway-stripe
# Account activation and currency requirements indicator

Addressed the account activation part of #2738 

## Changes proposed in this Pull Request:
- A banner is displayed on top of the payment methods section when the account has to be activated.
- An icon with a tooltip is displayed for each payment method when they require the store currency to be updated.
- Edited a folder name to `display-order-customization-notice.js` (unrelated to the original issue)

**Design**: [xVAk98kbRKUByVTrhPc98g-fi-641_71033](https://href.li/?https://www.figma.com/file/xVAk98kbRKUByVTrhPc98g/?node-id=641-71033)

## Testing instructions
- While in the develop branch, go to the Stripe settings page.
- Set `USD` as store currency.
- In the payment method list beside the method that requires activation, you should see a small yellow pill saying `Requires activation`
- In the payment method list beside the method that supports other currencies than `USD`, you should see a small yellow pill saying `Requires currency`

![before](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/1c60bb07-0ec6-441b-9cdc-35e751b9ded2)


- Now checkout to this branch and recheck the Stripe settings page.
- The `Requires activation` pill should not be there beside the individual payment methods.
- A banner should be displayed on top of the payment methods list indicating activation is required.
- Clicking on `Activate stripe account` button should take you to the Stripe dashboard.

![activation after](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/9e6972d3-8fa6-4b9b-add1-9e83e972c851)

- The payment methods that require other currencies i.e, `EUR` should not have the checkbox, instead it should have a yellow alert icon. Also the `Requires currency` pill should not be there beside the corresponding payment methods.
- Hovering on the tooltip should show the information about the required currency for the payment method.

![currency disabled](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/0ae0d46c-618d-447b-bffb-c8a2012b4072)
